darwin/future/data_objects/workflow.py

```diff
@@ -0,0 +1,172 @@
+from datetime import datetime
+from enum import Enum
+from typing import List, Optional
+from uuid import UUID
+
+from pydantic import Field
+
+from darwin.future.data_objects.typing import UnknownType
+from darwin.future.pydantic_base import DefaultDarwin
+
+
+class WFDataset(DefaultDarwin):
+    """
+    A class to manage all the information around a dataset on the darwin platform,
+    including validation
+
+    Attributes
+    ----------
+    id: int
+    name: str
+    instructions: str
+
+    Methods
+    ----------
+    __int__: returns the id of the dataset
+    __str__: returns the name of the dataset
+    __repr__: returns a string representation of the dataset
+    """
+
+    id: int
+    name: str
+    instructions: str = Field(min_length=0)
+
+    def __int__(self) -> int:
```

Comment:
I added these methods for ease, they may prove useful, they are better that running the Pydantic initial dunders in this case, and if they don't prove useful, they took very little time to do.